src/System.Management.Automation/engine/remoting/commands/EnterPSHostProcessCommand.cs

```diff
@@ -565,14 +574,45 @@ internal static IReadOnlyCollection<PSHostProcessInfo> GetAppDomainNamesFromProc
                                     }
                                 }
 
-                                int pNameIndex = namedPipe.IndexOf(".", pAppDomainIndex + 1, StringComparison.OrdinalIgnoreCase);
+                                int pNameIndex = namedPipe.IndexOf('.', pAppDomainIndex + 1);
                                 if (pNameIndex > -1)
                                 {
                                     string appDomainName = namedPipe.Substring(pAppDomainIndex + 1, (pNameIndex - pAppDomainIndex - 1));
                                     string pName = namedPipe.Substring(pNameIndex + 1);
 
-                                    procAppDomainInfo.Add(
-                                        new PSHostProcessInfo(pName, id, appDomainName));
+                                    Process process = null;
+
+                                    // Only return instances that aren't ourselves since you cannot enter yourself
```

Comment:
We should always return all processes that host PowerShell.  This cmdlet lists all processes that host PowerShell, not just processes that can be entered.  It is useful to get the complete list.

Reply:
While debugging, it was annoying to always check the current pid to know which I can actually enter.  Is there a real use case here where returning the process that you're current running `Get-PSHostProcessInfo` is useful since you can't do anything with it?

Reply:
Ok, it's fair for enumerating all PSHosts without intent to enter them all.  I'll defer this and just revert the filtering code.